ext/regex.go

```diff
@@ -330,3 +342,80 @@ func extractAll(target, regexStr ref.Val) ref.Val {
 	}
 	return types.NewStringList(types.DefaultTypeAdapter, result)
 }
+
+func estimateExtractCost() checker.FunctionEstimator {
+	return func(c checker.CostEstimator, target *checker.AstNode, args []checker.AstNode) *checker.CallEstimate {
+		if len(args) == 2 {
```

Comment:
Please add comments about what the logic below is doing, e.g.

```go
// compute cost as a function of the size of the target string multiplied by a traversal cost, times the size
// of the regex multiplied by a regex traversal cost. Ensure the resulting size is minimally zero and maximally
// the size of the target string.
...
```

Currently the result size is always the same as the target size, which I think is just the upper bound, but not the actual lower bound.